### ISSUE TITLE
CollectionOfTest.java:36-40: The test...

### ISSUE THREAD

**Issue by 0pdd** [yegor256/cactoos#902]
The puzzle `588-33083ad6` from #588 has to be resolved:

https://github.com/yegor256/cactoos/blob/a1e27f30945f217146ee23af795db7ae9da4850a/src/test/java/org/cactoos/collection/CollectionOfTest.java#L36-L40

The puzzle was created by rultor on 28-May-18. 

Estimate: 30 minutes,     

If you have any technical questions, don't ask me, submit new tickets instead. The task will be "done" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

**Comment by 0crat** [comment]
@llorllale[/z](https://www.0crat.com/u/llorllale) please, pay attention to this issue

**Comment by 0crat** [comment]
Job [#902](https://github.com/yegor256/cactoos/issues/902) is now in scope, role is `DEV`

**Comment by floreasorin** [comment]
This issues happens because an iterator can be iterated over only once. 
Before, there was only once check, now there are multiple checks in https://github.com/yegor256/cactoos/blob/a1e27f30945f217146ee23af795db7ae9da4850a/src/test/java/org/cactoos/collection/BehavesAsCollection.java#L67
and this gets called multiple times
https://github.com/yegor256/cactoos/blob/635ff72b524f88a76f7d14100d9b977fc60aeec1/src/main/java/org/cactoos/collection/CollectionOf.java#L75
The second time when it is called it will result in no values.

I suggest that to create a `ReapeatableIteratorOf(Iterator)` to use whenever collections need to be created out of iterators

**Comment by llorllale** [comment]
@floreasorin see #911 

The `CollectionOf(Iterator<E> iter)` ctor needs to be removed.

**Comment by 0crat** [comment]
The job [#902](https://github.com/yegor256/cactoos/issues/902) assigned to @danielpalafox[/z](https://www.0crat.com/u/danielpalafox), here is [why](https://www.0crat.com/footprint/C63314D6Z/1851864858); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be a monetary reward for this job

**Comment by DanielPalafox** [comment]
@0crat wait for reviewer

**Comment by 0pdd** [comment]
The puzzle `588-33083ad6` has disappeared from the source code, that's why I closed this issue.

**Comment by 0crat** [comment]
@elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova) please review this job completed by @danielpalafox[/z](https://www.0crat.com/u/danielpalafox), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#902](https://github.com/yegor256/cactoos/issues/902) is now out of scope

**Comment by elenavolokhova** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +35 point(s) just awarded to @danielpalafox[/z](https://www.0crat.com/u/danielpalafox)
